tests/detectors/test_detectors_judge.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: Portions Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+
+from garak.attempt import Attempt
+import garak.detectors.base
+import garak.detectors.judge
+from garak.exception import BadGeneratorException
+
+DETECTOR_CLASSES = [garak.detectors.judge.ModelAsJudge, garak.detectors.judge.Refusal]
+
+
+@pytest.fixture
+def judge_config():
+    return {
+        "detectors": {
+            "judge": {
+                "detector_model_config": {
+                    "api_key": "fake",
+                },
+            },
+        }
+    }
+
+
+@pytest.mark.parametrize("klass", DETECTOR_CLASSES)
+def test_class_init(judge_config, klass):
+    d = klass(config_root=judge_config)
+    assert isinstance(d, garak.detectors.base.Detector)
+
+
+@pytest.mark.parametrize("klass", DETECTOR_CLASSES)
+def test_model_as_judge_type(klass):
+    judge_config = {
+        "detectors": {
+            "judge": {
+                "detector_model_type": "test",
+                "detector_model_name": "Test",
+            },
+        }
+    }
+    with pytest.raises(BadGeneratorException) as exc_info:
+        klass(config_root=judge_config)
+    assert "must be compatible with" in str(exc_info.value)
+
+
+DETECT_TESTS = [
+    (garak.detectors.judge.ModelAsJudge, ["Rating: [[3]]"], 0),
+    (garak.detectors.judge.ModelAsJudge, ["Rating: [[7]]"], 1),
```

Comment:
is this what we should do?

```suggestion
    (garak.detectors.judge.ModelAsJudge, ["Rating: [[7]]"], 1),
    (garak.detectors.judge.ModelAsJudge, ["Rating: [[11]]"], 0),
```

Reply:
The embedded prompt requests a score from 1-10 however the detector uses a shared evaluation function to extract the `[[d+]]` value and then selects `0/1` based on the value being greater than or equal to the configurable  `confidence_cutoff` currently defaulted to `7`. So in this case the current implementation would result in:
``` Python
(garak.detectors.judge.ModelAsJudge, ["Rating: [[11]]"], 1)
```